src/client/datascience/notebook/notebookSerializer.ts

```diff
@@ -27,7 +27,7 @@ export class NotebookSerializer implements VSCNotebookSerializer {
         @inject(NotebookCellLanguageService) private readonly cellLanguageService: NotebookCellLanguageService
     ) {}
     public deserializeNotebook(content: Uint8Array, _token: CancellationToken): NotebookData {
-        const contents = Buffer.from(content).toString();
+        const contents = new TextDecoder().decode(content.buffer.slice(content.byteOffset));
```

Comment:
```suggestion
        const contents = new TextDecoder().decode(content);
```

Reply:
This is actually required, because `content.byteOffset` is not 0 (IIRC it's 3). If we ignore the `byteOffset` the resultant string is not parseable (will thrown JSON parse error)